controllers/cloud.redhat.com/providers/providers.go

```diff
@@ -16,6 +16,27 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type ProviderAccessor struct {
+	ProviderGet  func(c *Provider) (ClowderProvider, error)
```

Comment:
I feel like `SetupProvider` would be a better name here because it implies some more action beyond a simple fetch.